tests/application/cms/test_models.py

```diff
@@ -1,342 +1,361 @@
+        dimension = measure_version.dimensions[0]
+        dimension_guid = dimension.guid
+        original_chart_id = dimension.chart_id
+        original_table_id = dimension.table_id
 
         # The dimension has no table to start with
+        assert original_chart_id is not None
         assert original_table_id is None
 
         # Make a copy of a dimension and commit it to the DB
-        copied_dimension = dimension_fixture.copy()
+        copied_dimension = dimension.copy()
         db_session.session.add(copied_dimension)
         db_session.session.commit()
 
         # Re-fetch original dimension from db before asserting things about it
-        dimension_fixture = Dimension.query.filter(Dimension.guid == dimension_fixture_guid).one()
+        dimension = Dimension.query.filter(Dimension.guid == dimension_guid).one()
 
         # The copied dimension has a new Chart attached to it
         assert copied_dimension.dimension_chart is not None
         assert copied_dimension.chart_id != original_chart_id
         # The original dimension still has the old chart
-        assert dimension_fixture.chart_id == original_chart_id
+        assert dimension.chart_id == original_chart_id
         # The copied Chart is the same as the original Chart
-        assert copied_dimension.dimension_chart.classification_id == dimension_fixture.dimension_chart.classification_id
-        assert copied_dimension.dimension_chart.includes_all == dimension_fixture.dimension_chart.includes_all
-        assert copied_dimension.dimension_chart.includes_parents == dimension_fixture.dimension_chart.includes_parents
-        assert copied_dimension.dimension_chart.includes_unknown == dimension_fixture.dimension_chart.includes_unknown
+        assert copied_dimension.dimension_chart.classification_id == dimension.dimension_chart.classification_id
+        assert copied_dimension.dimension_chart.includes_all == dimension.dimension_chart.includes_all
+        assert copied_dimension.dimension_chart.includes_parents == dimension.dimension_chart.includes_parents
+        assert copied_dimension.dimension_chart.includes_unknown == dimension.dimension_chart.includes_unknown
 
         # Both the original and copied dimensions have no Table
-        assert dimension_fixture.dimension_table is None
+        assert dimension.dimension_table is None
         assert copied_dimension.table_id is None
         assert copied_dimension.dimension_table is None
 
-    def test_copy_dimension_also_copies_classification_links(self, db_session, stub_page_with_dimension_and_chart):
+    def test_copy_dimension_also_copies_classification_links(self, db_session):
         # Given a dimension with a classification link
-        dimension_fixture = Dimension.query.one()
-        dimension_fixture_guid = dimension_fixture.guid
+        measure_version = MeasureVersionWithDimensionFactory(
+            dimensions__classification_links__classification=ClassificationFactory(id="2A"),
```

Comment:
I think this could also be `dimensions__classification_links__classification__id="2A"`.